File: skyspaceeleven_ui/src/services/CustomerServiceResponseService.js
import http from "../http-common"; 

class CustomerServiceResponseService {
  getAllCustomerServiceResponses(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/customerServiceResponse/customerServiceResponses`, searchDTO);
  }

  get(customerServiceResponseId) {
    return this.getRequest(`/customerServiceResponse/${customerServiceResponseId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/customerServiceResponse?field=${matchData}`, null);
  }

  addCustomerServiceResponse(data) {
    return http.post("/customerServiceResponse/addCustomerServiceResponse", data);
  }

  update(data) {
  	return http.post("/customerServiceResponse/updateCustomerServiceResponse", data);
  }
  
  uploadImage(data,customerServiceResponseId) {
  	return http.postForm("/customerServiceResponse/uploadImage/"+customerServiceResponseId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CustomerServiceResponseService();

File: src/main/java/com/skyspaceeleven/service/CustomerServiceResponseService.java
package com.skyspaceeleven.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceeleven.domain.CustomerServiceResponse;
import com.skyspaceeleven.dto.CustomerServiceResponseDTO;
import com.skyspaceeleven.dto.CustomerServiceResponseSearchDTO;
import com.skyspaceeleven.dto.CustomerServiceResponsePageDTO;
import com.skyspaceeleven.dto.CustomerServiceResponseConvertCriteriaDTO;
import com.skyspaceeleven.service.GenericService;
import com.skyspaceeleven.dto.common.RequestDTO;
import com.skyspaceeleven.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CustomerServiceResponseService extends GenericService<CustomerServiceResponse, Integer> {

	List<CustomerServiceResponse> findAll();

	ResultDTO addCustomerServiceResponse(CustomerServiceResponseDTO customerServiceResponseDTO, RequestDTO requestDTO);

	ResultDTO updateCustomerServiceResponse(CustomerServiceResponseDTO customerServiceResponseDTO, RequestDTO requestDTO);

    Page<CustomerServiceResponse> getAllCustomerServiceResponses(Pageable pageable);

    Page<CustomerServiceResponse> getAllCustomerServiceResponses(Specification<CustomerServiceResponse> spec, Pageable pageable);

	ResponseEntity<CustomerServiceResponsePageDTO> getCustomerServiceResponses(CustomerServiceResponseSearchDTO customerServiceResponseSearchDTO);
	
	List<CustomerServiceResponseDTO> convertCustomerServiceResponsesToCustomerServiceResponseDTOs(List<CustomerServiceResponse> customerServiceResponses, CustomerServiceResponseConvertCriteriaDTO convertCriteria);

	CustomerServiceResponseDTO getCustomerServiceResponseDTOById(Integer customerServiceResponseId);







}





